2.MLP_Multi_Classifier/MLP_utils/utils.py

```diff
@@ -0,0 +1,938 @@
+"""
+Functions for Machine Learning Model optimization, training and testing.
+These are helper functions meant to be called in a separate notebook or script
+"""
+
+
+import ast
+from configparser import ConfigParser
+from typing import Tuple
+
+import matplotlib.pyplot as plt
+import numpy as np
+import optuna
+import pandas as pd
+import seaborn as sns
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from MLP_utils.parameters import Parameters
+from sklearn.metrics import (
+    accuracy_score,
+    auc,
+    classification_report,
+    confusion_matrix,
+    f1_score,
+    precision_score,
+    recall_score,
+    roc_auc_score,
+    roc_curve,
+)
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import label_binarize
+
+config = ConfigParser()
+config.optionxform = str
+config.read("MLP_utils/config.ini")
+
+params = Parameters()
+
+
+def parameter_set(params: Parameters, config: object) -> object:
+    """reset parameter class defaults by updating from config
+
+    Parameters
+    ----------
+    params : object
+        param class holding parameter information
+    config: object
+        config class
+
+    Returns
+    -------
+    object
+        param class holding updated parameter information
+    """
+    params.DATA_SUBSET_OPTION = config["DEFAULT"]["DATA_SUBSET_OPTION"]
+    params.DATA_SUBSET_NUMBER = int(config["DEFAULT"]["DATA_SUBSET_NUMBER"])
+    params.BATCH_SIZE = int(config["DEFAULT"]["BATCH_SIZE"])
+    params.OPTIM_EPOCHS = int(config["DEFAULT"]["OPTIM_EPOCHS"])
+    params.N_TRIALS = int(config["DEFAULT"]["N_TRIALS"])
+    params.TRAIN_EPOCHS = int(config["DEFAULT"]["TRAIN_EPOCHS"])
+    params.MIN_LAYERS = int(config["DEFAULT"]["MIN_LAYERS"])
+    params.MAX_LAYERS = int(config["DEFAULT"]["MAX_LAYERS"])
+    params.LAYER_UNITS_MIN = int(config["DEFAULT"]["LAYER_UNITS_MIN"])
+    params.LAYER_UNITS_MAX = int(config["DEFAULT"]["LAYER_UNITS_MAX"])
+    params.DROPOUT_MIN = float(config["DEFAULT"]["DROPOUT_MIN"])
+    params.DROPOUT_MAX = float(config["DEFAULT"]["DROPOUT_MAX"])
+    params.DROP_OUT_STEP = float(config["DEFAULT"]["DROP_OUT_STEP"])
+    params.LEARNING_RATE_MIN = float(config["DEFAULT"]["LEARNING_RATE_MIN"])
+    params.LEARNING_RATE_MAX = float(config["DEFAULT"]["LEARNING_RATE_MAX"])
```

Comment:
Seeing how these params are casted to specific data types, would there be an opportunity to use dataclass type specifications? ([see this reference](https://docs.python.org/3/library/dataclasses.html#dataclasses.field))

Reply:
@d33bs Yes, there is and I have incorporated it in the `Parameters.py` filebut unforntunatly the parser retruns `str` type and I must type cast these into `int` or `float`.

Reply:
Sorry to hear, I understand, thanks!